src/thingsboard.c

```diff
@@ -22,6 +24,7 @@ static struct {
 K_SEM_DEFINE(time_sem, 0, 1);
 
 static attr_write_callback_t attribute_cb;
+static rpc_callback_t rpc_cb;
```

Comment:
The callback will be overwritten by another one if the user calls `thingsboad_rpc` multiple times without waiting for the response. When the response arrives the wrong callback would be called.
Avoiding multiple RPCs in flight with a semaphore approach that you mentioned yesterday would probably solve this. Or just reset `rpc_cb` to `NULL` after it was called and return with an error from `thingsboad_rpc()` if `rpc_cb` is not `NULL`?

A nicer solution would be to allow simultaneous RPC calls by using a memory slab to store multiple callbacks (the same way [coap_client.c manages multiple requests](https://github.com/grandcentrix/thingsboard-sdk/blob/main/src/coap_client.c#L19)).

Reply:
yeah, the limitation is obvious. I will think about a solution, being able to do multiple RPC calls simultaneously seems like something that's not often used, but could come in handy.

Reply:
I would be fine with having this limitations as it is probably okay for most use cases. We should just document this and somehow enforce the "only one rpc at a time" rule to avoid weird behaviour (like wrong callback called) for the user.